docs/sbpy/data/dataclass.rst

```diff
@@ -477,16 +472,42 @@ Similarly, existing columns can be modified using:
 
     >>> obs['filter'] = ['g', 'i', 'R', 'V']
 
-Note how the `~astropy.table.Table.add_column` and
-`~astropy.table.Table.add_row` functions are called from
-``obs.table``. `~sbpy.data.DataClass.table` is a property that exposes
-the underlying `~astropy.table.QTable` object so that the user can
-directly interact with it. Please refer to the `~astropy.table.Table`
-reference and
-[documentation](https://docs.astropy.org/en/stable/table/index.html)
-for more information on how to modify `~astropy.table.QTable` objects.
+If you want to join two observations into a single object:
 
+.. doctest-requires:: astropy>=5
 
+    >>> ra = [20.223423, 20.233453, 20.243452] * u.deg
+    >>> dec = [12.42123, 12.41562, 12.40435] * u.deg
+    >>> phase = [10.1, 12.3, 15.6] * u.deg
+    >>> epoch = Time(2451623.5 + array([0.1234, 0.2345, 0.3525]), format='jd')
+    >>> obs2 = Obs.from_columns([ra, dec, epoch, phase],
+    ...     names=['ra', 'dec', 't', 'phase'])
+    >>>
+    >>> obs.join(obs2)
+    >>> obs
+    <QTable length=7>
+        ra       dec          t       filter filter2  phase
+       deg       deg                                   deg
+     float64   float64       Time      str1    str1  float64
+    --------- --------- ------------- ------ ------- -------
+    10.323423 -12.42123  2451523.6234      g       V     ———
+    10.333453 -12.41562  2451523.7345      i       V     ———
+    10.343452 -12.40435  2451523.8525      R       R     ———
+     10.25546  -12.3946 2451523.94653      V       i     ———
+    20.223423  12.42123  2451623.6234     --      --    10.1
+    20.233453  12.41562  2451623.7345     --      --    12.3
+    20.243452  12.40435  2451623.8525     --      --    15.6
+
+Note that the data table to be joined doesn't have to have the same
+columns as the original data table.  The empty field will be automatically
+masked out.
```

Comment:
Can we have this as an option that can be turned off?  Or maybe it is off by default?  As a programmer, I can see some value in having an exception raised when the columns do not match.

Reply:
Since this method is basically equivalent to `astropy.table.Table.vstack()` (as you correctly point out), I'll just use the 'join_type' keyword parameter to deal with the different cases.

`vstack` has default join_type as 'outer', which means merge with all columns and mask out the cells with no values.  I think it's a good idea to maintain the consistency, and just let `Table.vstack()` raise any necessary exceptions.

Sounds good to you?

Reply:
Ah, yes, that sounds good.